akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -65,6 +71,19 @@ import akka.actor.typed.Behavior.UntypedBehavior
     val ref = cell.addFunctionRef((_, msg) ⇒ untyped.self ! AdaptMessage[U, T](msg.asInstanceOf[U], f), _name)
     ActorRefAdapter[U](ref)
   }
+
+  override def log: ActorLogger = {
+    actorLogger match {
+      case OptionVal.Some(logger) ⇒ logger
+      case OptionVal.None ⇒
+        import scala.language.existentials
+        val (str, clazz) = LogSource(untyped.self) // FIXME clazz doesn't really make as much sense for typed, what to do?
```

Comment:
for MutableBehavior it does

Reply:
I meant more, that the untyped self wont give us much useful info for the class here, so should rather be something from the typed behavior then.

Reply:
Looking at it some more, we'd have to reach out to the `ActorAdapter` and its' `initialBehavior` somehow (which I don't quite see how to do), and even then it will likely be our behavior types that we mostly get out of it.